packages/settingeditor/src/pluginlist.tsx

```diff
@@ -237,10 +250,112 @@ export class PluginList extends ReactWidget {
     return typeof hint === 'string' ? hint : '';
   }
 
+  /**
+   * Function to recursively filter properties that match search results.
+   * @param filter - Function to filter based on search results
+   * @param props - Schema properties being filtered
+   * @param definitions - Definitions to use for filling in references in properties
+   * @param ref - Reference to a definition
+   * @returns - String array of properties that match the search results.
+   */
+  getFilterString(
+    filter: (item: string) => boolean,
+    props: ISettingRegistry.IProperty,
+    definitions?: any,
+    ref?: string
+  ): string[] {
+    // If properties given are references, populate properties
+    // with corresponding definition.
+    if (ref && definitions) {
+      ref = ref.replace('#/definitions/', '');
+      props = definitions[ref] ?? {};
+    }
+
+    // If given properties are an object, advance into the properties
+    // for that object instead.
+    if (props.properties) {
+      props = props.properties;
+      // If given properties are an array, advance into the properties
+      // for the items instead.
+    } else if (props.items) {
+      props = props.items as any;
+      // Otherwise, you've reached the base case and don't need to check for matching properties
+    } else {
+      return [];
+    }
+
+    // If reference found, recurse
+    if (props['$ref']) {
+      return this.getFilterString(
+        filter,
+        props,
+        definitions,
+        props['$ref'] as string
+      );
+    }
+
+    // Make sure props is non-empty before calling reduce
+    if (Object.keys(props).length === 0) {
+      return [];
+    }
+
+    // Iterate through the properties and check for titles / descriptions that match search.
+    return reduce(
+      Object.keys(props),
```

Comment:
You could use `Array.reduce` directly

![image](https://user-images.githubusercontent.com/8435071/154654944-4019a3b1-35a6-4f17-8821-07659ac1a11e.png)

```suggestion
    // Iterate through the properties and check for titles / descriptions that match search.
    return Object.keys(props).reduce(
```